server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileServicesImpl.java

```diff
@@ -237,6 +217,47 @@ public List<MemberProfile> getSubordinatesForId(UUID id) {
     @Override
     @CacheInvalidate(cacheNames = {"member-cache"})
     public MemberProfile updateProfile(MemberProfile memberProfile) {
+        if (memberProfile.getId() == null) {
+            throw new BadArgException("Null profile id in update");
+        }
+
+        MemberProfile currentUser = currentUserServices.getCurrentUser();
+        boolean isAdmin = currentUserServices.isAdmin();
+        if (!isAdmin && (currentUser == null || !currentUser.getId().equals(memberProfile.getId()))) {
+             throw new PermissionException(NOT_AUTHORIZED_MSG);
+        }
+
+        MemberProfile emailProfile = memberProfileRepository.findByWorkEmail(memberProfile.getWorkEmail()).orElse(null);
+
+        if (emailProfile != null && emailProfile.getId() != null && !Objects.equals(memberProfile.getId(), emailProfile.getId())) {
+            throw new AlreadyExistsException(String.format("Email %s already exists in database",
+                    memberProfile.getWorkEmail()));
+        }
+
+        Optional<MemberProfile> existingProfileOpt = memberProfileRepository.findById(memberProfile.getId());
+        MemberProfile updatedMemberProfile = memberProfileRepository.update(memberProfile);
+        if (existingProfileOpt.isEmpty()) {
+            LOG.error("MemberProfile with id {} not found", memberProfile.getId());
+        } else {
+            MemberProfile existingProfile = existingProfileOpt.get();
+
+            boolean pdlChanged = !Objects.equals(existingProfile.getPdlId(), memberProfile.getPdlId());
+            boolean supervisorChanged = !Objects.equals(existingProfile.getSupervisorid(), memberProfile.getSupervisorid());
+
+            if (pdlChanged) {
+                emailAssignment(updatedMemberProfile, true); // PDL
+            }
+            if (supervisorChanged) {
+                emailAssignment(updatedMemberProfile, false); // Supervisor
+            }
+        }
+
+        return updatedMemberProfile;
+    }
+
+    @Override
+    @CacheInvalidate(cacheNames = {"member-cache"})
+    public MemberProfile updateCurrentUserProfile(MemberProfile memberProfile) {
```

Comment:
This seems to be misleading. This isn't related to the "current user" at all, is it? Am I missing something?

Reply:
Well, it is only called from the current user controller when updating the "last seen" for the current user.

Reply:
Maybe it should be renamed to something like "directProfileUpdate" or something indicating that there is no security related to it?

Reply:
Maybe we isolate that code to update the "last seen" here and call it something like 'updateLastSeen' and just pass in the user id? that feels less risky than an unsecured profile update API.